### PR TITLE
Code to reproduce JOSS figure

### PR DESCRIPTION
**Pull request by FLomb**
JOSS's reviewers asked for the inclusion of the code to reproduce the illustrative Figure we have in the paper (see: [https://github.com/openjournals/joss-reviews/issues/6418#issuecomment-1999853280](https://github.com/openjournals/joss-reviews/issues/6418#issuecomment-1999853280)). This PR addresses this.

However, the PR is currently based on the old plotting functionality with some customisation. As discussed during RAMP's development meetings, it'd be better to produce a similar Figure with the newest plotting functionality. I tried doing so via the `shadow` plot functionality, but I struggled to customise it to the degree needed. For example, I did not manage to pass kwargs related to the ax object or to the colours of the lines. Perhaps @mohammadamint may know how to do it. If not possible, we may as well stick to this custom code.

### CODE REVIEW

File: docs/notebooks/plot_class.ipynb
Comment (FLomb) [inline docs/notebooks/plot_class.ipynb:1]: I'm not sure why this file ended up in the commit; I shouldn't have changed anything there and for some reason `git diff` does not return any difference between my commit and the `joss-paper` branch for this file.

Comment (Bachibouzouk) [reply docs/notebooks/plot_class.ipynb:1]: Apparently the image generated must be slightly different, this is the only line highlighted (once you commited `git diff` will not show the diff anymore), you can do `git diff HEAD~1` to get the diff of the last commit.

**Comment by Bachibouzouk** [comment]
You can use 
```
uc = UseCase("JOSS-paper-figure", users=[LMI, LI])
uc.initialize(num_days=1)
plot = uc.generate_daily_load_profiles(
    cases=[1,2,3,4,5]
)
plt.rcParams.update({'font.size': 22})

fig, ax1 = plot.shadow( figsize=(16, 8))
ax1.legend(['5-day average', 'Single days(s)'])
ax1.margins(0)

ax1.set_xticks(ax1.get_xticks()[1:])
import matplotlib.dates as mdates
myFmt = mdates.DateFormatter('%H:%M')
ax1.xaxis.set_major_formatter(myFmt)

ax1.set_ylabel('Normalised power (kW/kW-Peak)', fontsize='large', labelpad=20)
ax1.set_xlabel('Time (hours:minutes)', fontsize='large', labelpad=20)
```

To generate a figure similar using the new plotting capabilities

**Comment by mohammadamint** [comment]
> You can use
> 
> ```
> uc = UseCase("JOSS-paper-figure", users=[LMI, LI])
> uc.initialize(num_days=1)
> plot = uc.generate_daily_load_profiles(
>     cases=[1,2,3,4,5]
> )
> plt.rcParams.update({'font.size': 22})
> 
> fig, ax1 = plot.shadow( figsize=(16, 8))
> ax1.legend(['5-day average', 'Single days(s)'])
> ax1.margins(0)
> 
> ax1.set_xticks(ax1.get_xticks()[1:])
> import matplotlib.dates as mdates
> myFmt = mdates.DateFormatter('%H:%M')
> ax1.xaxis.set_major_formatter(myFmt)
> 
> ax1.set_ylabel('Normalised power (kW/kW-Peak)', fontsize='large', labelpad=20)
> ax1.set_xlabel('Time (hours:minutes)', fontsize='large', labelpad=20)
> ```
> 
> To generate a figure similar using the new plotting capabilities

@Bachibouzouk Thanks :-). Sorry for my delay in replying!

**Comment by FLomb** [comment]
Ok, thanks for your guidance. I've updated the figure accordingly. What we get is an almost identical result, except for colours. I reckon there's no way to change the colours of the `shadow` function because they are hardcoded, aren't they? Should I commit a change to the function that allows for colour customisation, too?

**Comment by Bachibouzouk** [comment]
> function because they are hardcoded, aren't they?

You can get the plotted lines from the axis handle
https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.axes.Axes.get_lines.html

**Comment by FLomb** [comment]
> > function because they are hardcoded, aren't they?
> 
> You can get the plotted lines from the axis handle https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.axes.Axes.get_lines.html

Thanks for the tip! I've updated the code with custom colours. Now, the figure looks as expected; I should update the paper with it because it's still marginally different.

**Review by Bachibouzouk — APPROVED** [review]
Thanks @FLomb this looks neat

I added a seed to the UseCase so the figure is reproducable

**Pull request merged**